value/src/main/java/com/google/auto/value/processor/AnnotationOutput.java

```diff
@@ -0,0 +1,244 @@
+      this.context = context;
+    }
+
+    @Override
+    public Void visitAnnotation(AnnotationMirror a, StringBuilder sb) {
+      processingEnv.getMessager().printMessage(
+          Diagnostic.Kind.ERROR,
+          "@AutoAnnotation cannot yet supply a default value for annotation-valued member '"
+              + memberName + "'",
+          context);
+      sb.append("null");
+      return null;
+    }
+  }
+
+  private class AnnotationSourceFormVisitor extends SourceFormVisitor {
+    @Override
+    public Void visitAnnotation(AnnotationMirror a, StringBuilder sb) {
+      sb.append('@').append(typeSimplifier.simplify(a.getAnnotationType()));
+      Map<ExecutableElement, AnnotationValue> map = ImmutableMap.copyOf(a.getElementValues());
+      if (!map.isEmpty()) {
+        sb.append('(');
+        String sep = "";
+        for (Map.Entry<ExecutableElement, AnnotationValue> entry : map.entrySet()) {
+          sb.append(sep).append(entry.getKey().getSimpleName()).append(" = ");
+          sep = ", ";
+          this.visit(entry.getValue(), sb);
+        }
+        sb.append(')');
+      }
+      return null;
+    }
+  }
+
+  /**
+   * Returns a string representation of the given annotation value, suitable for inclusion in a Java
+   * source file as the initializer of a variable of the appropriate type.
+   */
+  String sourceFormForInitializer(
+      AnnotationValue annotationValue,
+      ProcessingEnvironment processingEnv,
+      String memberName,
+      Element context) {
+    SourceFormVisitor visitor =
+        new InitializerSourceFormVisitor(processingEnv, memberName, context);
+    StringBuilder sb = new StringBuilder();
+    visitor.visit(annotationValue, sb);
+    return sb.toString();
+  }
+
+  /**
+   * Returns a string representation of the given annotation mirror, suitable for inclusion in a
+   * Java source file to reproduce the annotation in source form.
+   */
+  String sourceFormForAnnotation(AnnotationMirror annotationMirror) {
+    StringBuilder sb = new StringBuilder();
+    new AnnotationSourceFormVisitor().visitAnnotation(annotationMirror, sb);
+    return sb.toString();
+  }
+
+  private static StringBuilder appendQuoted(StringBuilder sb, String s) {
+    sb.append('"');
+    for (int i = 0; i < s.length(); i++) {
+      appendEscaped(sb, s.charAt(i));
+    }
+    return sb.append('"');
+  }
+
+  private static StringBuilder appendQuoted(StringBuilder sb, char c) {
+    sb.append('\'');
+    appendEscaped(sb, c);
+    return sb.append('\'');
+  }
+
+  private static void appendEscaped(StringBuilder sb, char c) {
+    switch (c) {
+    case '\\':
+    case '"':
+    case '\'':
+      sb.append('\\').append(c);
+      break;
+    case '\n':
+      sb.append("\\n");
+      break;
+    case '\r':
+      sb.append("\\r");
+      break;
+    case '\t':
+      sb.append("\\t");
+      break;
+    default:
+      if (c < 0x20) {
+        sb.append(String.format("\\%03o", (int) c));
```

Comment:
I guess \u escapes would work too, since we've already eliminated the characters (\n and \r) that could cause problems, given that \u escapes are replaced before string constants are parsed.